### PR TITLE
Add react-native-web to the template

### PR DESCRIPTION
**Pull request by raarts**
`yarn web` will now start webpack-dev-server, which will show the same screen as the ios/android apps.
`yarn build` will build for production (into public/assets).

Note this install `expo-web` as a dependency, and  `babel-plugin-expo-web` to replace all instances of `import { .... } from 'expo'` with `import { .... } from 'expo-web'`.

### CODE REVIEW

File: react-native-scripts/src/scripts/init.js [hunk 1]
```diff
@@ -71,6 +86,8 @@ https://github.com/npm/npm/issues/16991
     eject: 'react-native-scripts eject',
     android: 'react-native-scripts android',
     ios: 'react-native-scripts ios',
+    web: './node_modules/.bin/webpack-dev-server -d --config ./webpack.config.js  --inline --hot --colors --content-base public/',
```
Comment (brentvatne) [inline react-native-scripts/src/scripts/init.js]: no need to scope with `./node_modules` here, we can just do `webpack-dev-server`

**Comment by brentvatne** [comment]
the direction of this looks great. one request which may complicate things a bit but I think is important to keep this as an opt-in feature until it stabilizes — could we possibly put it under a flag? so `create-react-native-app raarts-app --with-web-support` for example would be amazing

**Comment by brentvatne** [comment]
ideally we would hide the webpack config but I think web support isn't at a level of maturity where we can get away with that yet. I'm going to land this and publish it in a pre-release so more people can play with it. thanks!

**Comment by necolas** [comment]
FWIW I think Metro is supposed to support bundling for web now (and code-splitting at some point). So you might be able to rely on the same bundler for all platforms pretty soon.

**Comment by sanderlooijenga** [comment]
After the creating the native app with this argument, and ejecting it, I do have some problems when running `npm run ios`, due `error: bundling failed: Error: Unable to resolve module `./App` from …`. Did somebody else have this issue as well?

**Pull request merged**